modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor_balloon.jsp

```diff
@@ -22,6 +22,10 @@ Map<String, Object> editorData = (Map<String, Object>)request.getAttribute(CKEdi
 String name = namespace + GetterUtil.getString((String)request.getAttribute(CKEditorConstants.ATTRIBUTE_NAMESPACE + ":name"));
 %>
 
+<liferay-util:html-top>
```

Comment:
This means that the editor can't be used directly from JS which is one of the 2 main use cases we have. I don't think we want to go back to this, which makes using the component really tricky as we've seen in the past.

Reply:
Good point @jbalsas. Do you have any idea why the import in the BalloonEditor react component affects other editors? or how we can fix it?

Reply:
No idea, that's for you to figure out! 😉

Reply:
Just for the record:  after speaking with @izaera , we found that importing a css in a component applies to all the module as it is something done in build time, and not in runtime. So we had to look for a different solution, restricting the css to the inline editor: https://github.com/liferay-frontend/liferay-portal/pull/964/commits/3ac66beb5d371eb0a24f0da22033f5eafbe4f790

Reply:
I don't think that's the case. Could you try commenting out [this line](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/index.js#L15) and see what you can infer from the results?

Reply:
Sure, I'll check that right now.

Reply:
What I'm seeing when the  [`BalloonEditor` `export`](https://github.com/liferay-frontend/liferay-portal/blob/850fb8025300f4e449314e98ec97c4bfd3bf06c0/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/index.js#L15) is commented is the following:

- `BalloonEditor` is not "required" in the `index.js` file which looks like [this](https://gist.github.com/julien/e210a41c25b99c0eb3c6dd9d3810ec42) 
 
-  The `main.scss` file is present in the jar, along with a `main.scss.js` that looks like [this](https://gist.github.com/julien/3842cd6ea1eec8efae56897f5d2cc189) 

When, the `BalloonEditor` `export` is not commented (right now):

- `BalloonEditor` is required in the `index.js` file which looks like [this](https://gist.github.com/julien/4758826c3eaae5d13f857e89a4a6eb3c). (Here's a [diff](https://gist.github.com/julien/30cccbe606ddfd98d6b209165969c73a) of the `index.js` file showing what's added when `BalloonEditor` is exported)

- the `main.scss` file is included in the jar (as expected) with `main.scss.js`, that looks exactly the same

- A `<script>` tag including `main.scss.js` is added to the page (which isn't the case when `BalloonEditor` isn't exported from `index.js`)

![2021-04-12-14:59:52](https://user-images.githubusercontent.com/5572/114398198-cf013400-9b9f-11eb-8a2d-78e4791ebb6f.png)
 
Based on that, (and I don't know if it's possible) it would be nice to be able to avoid exporting `BalloonEditor` from `index.js`, but to be able to `import` it from elsewhere.

Reply:
> Based on that, (and I don't know if it's possible) it would be nice to be able to avoid exporting BalloonEditor from index.js, but to be able to import it from elsewhere.

It's not.

> When, the BalloonEditor export is not commented (right now):

I'm missing what's causing those files to be included from the analysis. You've described what you observe is happening (and the thing that's causing it) but not why.

Brian needs to merge this to trigger a new GA1 publication asap. The way I see it we can:
- Send a full revert for today
- Send a single commit commenting out the editor for today
- Add a specific class to the editor to properly scope the css
- Other...?

@julien, @ambrinchaudhary, can you pick one of the options and move with it as soon as we can?

Reply:
> I'm missing what's causing those files to be included from the analysis. You've described what you observe is happening (and the thing that's causing it) but not why.

Yes that's true, but I don't understand (yet) why that file is being included.

Concerning the release I'd say that 

> Send a single commit commenting out the editor for today

Would probably be the fastest way, the inline (floating) toolbar is rendered outside the component (in the document's body) so finding the correct CSS rule is going to take some time

Reply:
> > Based on that, (and I don't know if it's possible) it would be nice to be able to avoid exporting BalloonEditor from index.js, but to be able to import it from elsewhere.
> 
> It's not.

Technically it is possible, just not "The Recommended Way" to do things.

- `index.js` depends on `BalloonEditor.js`, so you can't load `index.js` without loading `BalloonEditor.js`.
- `BalloonEditor.js` depends on the SCSS, so you can't load _that_ without loading the SCSS.
- Therefore, you can't load `index.js` without loading the SCSS.

Now, if we drop `BalloonEditor` from the `index.js`, we no longer get the SCSS (as you saw). And its still possible for others to load it by importing from `frontend-editor-ckeditor-web/editor/BalloonEditor`. But it's not "The Recommended Way" because:

- We want the `index.js` to be the "manifest" of the public API of the project. Reaching into internals is breaking encapsulation and assuming that file paths won't change, internal names won't change etc.
- Given the global nature of CSS, even if you only load the bad SCSS when somebody explicitly asks for the balloon editor, we could _still_ wind up breaking things quite easily (if there is a non-balloon editor on the page).

> - Add a specific class to the editor to properly scope the css

This sounds like the "best" solution.

> - Send a full revert for today

This sounds like the most direct way to mitigate the risk immediately. I'd do this.

Reply:
The problem with delaying when the CSS is loaded is what I told @julien, I think... 

If you delay the load you are simply delaying the error, but the result will be the same. If the problem is that loading a CSS file breaks things in a different place because of cascading, the same thing will happen if you load the CSS file later, once it is loaded.

If I understood correctly.

If I'm wrong and delaying the load makes it, there's another option: using `require` instead of `import`, which allows wiring modules at runtime. That's supported by our bundler and by webpack (and I would say any other bundling tool).

Reply:
Note I'm not recommending to use `require` in normal scenarios, but since @wincent proposed to deviate from The Correct Way :tm:, I wanted to add my 2 cents to the pool of heretical proposals.

Reply:
> since @wincent proposed

Definitely not proposing it; just saying it's possible (and why I wouldn't do it).